scripts/crowdin/pretranslate.mjs

```diff
@@ -0,0 +1,159 @@
+const projectId = process.env.CROWDIN_PROJECT_ID;
+const token =  process.env.CROWDIN_PERSONAL_TOKEN;
+
+const BASE_URL = `https://api.crowdin.com/api/v2/projects/${projectId}`;
+const preTranslateURL = `${BASE_URL}/pre-translations`;
+const REQUEST_INTERVAL_TIMEOUT = 10_000;
+const MAXIMUM_REQUEST_RETRIES = 12;
+
+const requestData = {
+  method: 'mt',
+  engineId: 410374,
+  autoApproveOption: 'all',
+  duplicateTranslations: false,
+  skipApprovedTranslations: true,
+  translateUntranslatedOnly: true,
+  translateWithPerfectMatchOnly: false,
+};
+
+const getMachineTranslationEngineID = async () =>{
+  try {
+    const response = await fetch('https://api.crowdin.com/api/v2/mts', {
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+    });
+
+    if (!response.ok) {
+      throw new Error(`Failed to get target language ids`);
+    }
+
+    const responseData = await response.json();
+    const engineId = responseData.data[0].data.id ;
+
+    return engineId;
+
+  } catch (error) {
+    console.error('Error:', error);
+    throw error;
+  }
+}
+
+const getLanguageIds = async () => {
+  try {
+    const response = await fetch(`${BASE_URL}`, {
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+    });
+
+    if (!response.ok) {
+      throw new Error(`Failed to get target language ids`);
+    }
+
+    const responseData = await response.json();
+    const languageIds = responseData.data.targetLanguageIds;
+
+    return languageIds;
+  } catch (error) {
+    console.error('Error:', error);
+    throw error;
+  }
+};
+
+const getSourceFileId = async () => {
+  try {
+    const response = await fetch(`${BASE_URL}/files`, {
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+    });
+
+    if (!response.ok) {
+      throw new Error(`Failed to get source file id`);
+    }
+
+    const responseData = await response.json();
+    const sourceFileId = responseData.data[0].data.id;
+
+    return sourceFileId;
+  } catch (error) {
+    console.error('Error:', error);
+    throw error;
```

Comment:
Use same approach as we did in `scripts/common/errors.mjs` for errors. You can add CrowdinError for instance and use it for your errors here.

Reply:
Same comment applies to other errors.

Reply:
FIxed